test/dlc_tests/container_tests/bin/benchmark/read_upload_benchmarking_results.py

```diff
@@ -0,0 +1,75 @@
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+import datetime
+import pandas as pd
+import os
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-dir", "--directory", type=str)
+
+
+def get_boto3_session(region="us-east-1"):
+    """Get boto3 session with us-east-1 as default region used to connect to AWS services."""
+    return boto3.session.Session(region_name=region)
+
+def get_cloudwatch_client(region="us-east-1"):
+    """Get AWS CloudWatch client object. Currently assume region is IAD (us-east-1)"""
+    return get_boto3_session(region=region).client("cloudwatch")
+
+def put_metric_data(metric_name, namespace, unit, value, dimensions):
+    """Puts data points to cloudwatch metrics"""
+    cloudwatch_client = get_cloudwatch_client()
+    current_timestamp = datetime.datetime.utcnow()
+    try:
+        response = cloudwatch_client.put_metric_data(
+            Namespace=namespace,
+            MetricData=[
+                {
+                    "MetricName": metric_name,
+                    "Dimensions": dimensions,
+                    "Value": value,
+                    "Unit": unit,
+                    "Timestamp": current_timestamp,
+                }
+            ],
+        )
+    except ClientError as e:
+        LOGGER.error("Error: Cannot put data to cloudwatch metric: {}".format(metric_name))
+        LOGGER.error("Exception: {}".format(e))
+        raise e
+
+def read_metric(csv_file):
+    df = pd.read_csv(csv_file)
+    value = df[df.columns[-1]].iloc[0]
+    if isinstance(value, str):
+        for i in range(len(value)):
+            if value[i].isdecimal() or value[i] == ".":
+                continue
+            else:
+                return float(value[:i])
+    return float(value)
+
+def upload_metric(unique_cw_namespace, value, unique_metric_dims):
```

Comment:
Normal functions and pytest fixtures work differently. Please rework this function to actually upload the metrics to CW instead of returning a callable.